tests/unit/src/main/scala/tests/BaseLspSuite.scala

```diff
@@ -77,6 +85,21 @@ abstract class BaseLspSuite(
     assertNoDiff(obtained, expectedName)
   }
 
+  def test(testOpts: TestOptions, withoutVirtualDocs: Boolean)(
+      fn: => Future[Unit]
+  )(implicit loc: Location) {
+    if (withoutVirtualDocs) {
+      test(testOpts.withName(s"${testOpts.name}-readonly")) { fn }
+      test(
+        testOpts
+          .withName(s"${testOpts.name}-virtualdoc")
+          .withTags(Set(virtualDocTag))
+      ) { fn }
+    } else {
+      test(testOpts)(fn)
+    }
+  }
+
   def newServer(workspaceName: String): Unit = {
```

Comment:
We could add an additional option to `newServer` 
```suggestion
  def newServer(workspaceName: String, useVirtualDocs): Unit = {
```
and check in beforeEach if the tag is set, which would probably be better than the var I proposed.

Reply:
The var is still needed as the tests themselves need to know whether they are using virtualdocs, so need to be able to call `BaseLspSuite#useVirtualDocuments`.  And it can't just interrogate `initializationOptions` as this doesn't contain the virtualdoc info as the `initializationOptions` are added to in `newServer`.  We could add the additional option to `newServer` but it would just be duplicating the `BaseLspSuite#useVirtualDocs` var

Reply:
Ok, makes sense!